src/Permit2Relayer.sol

```diff
@@ -69,4 +72,73 @@ contract Permit2Relayer {
 
         IERC20(permit.permitted.token).safeTransfer(recipient, permit.permitted.amount);
     }
+
+    function requestCrossChainSwapPermit2(
+        address router,
+        address signer,
+        address tokenIn,
+        address tokenOut,
+        uint256 amount,
+        uint256 solverFee,
+        uint256 dstChainId,
+        address recipient,
+        IPermit2.PermitTransferFrom memory permit,
+        bytes calldata signature,
+        bytes calldata additionalData
+    ) external {
+        /// @notice Type string of the custom witness
+        string memory witnessTypeString = string(
+            abi.encodePacked(
+                WITNESS_TYPE_NAME,
+                "(address router,address tokenIn,address tokenOut,uint256 amount,uint256 solverFee,uint256 dstChainId,address recipient,bytes additionalData)"
+            )
+        );
+
+        /// @notice Type hash used to compute the witness hash
+        bytes32 witnessTypeHash = keccak256(bytes(witnessTypeString));
+
+        /// @notice The permit2 witnessTypeString parameter,
+        string memory permit2WitnessTypeString = string(
+            abi.encodePacked(
+                WITNESS_TYPE_NAME, " witness)", witnessTypeString, "TokenPermissions(address token,uint256 amount)"
+            )
+        );
```

Comment:
Could make those parameters constant to save some gas. I'd probably also use a different `WITNESS_TYPE_NAME` for the `crossChainSwap` function.